1-development/list-posts-by-rating/src/main/java/com/Joao_Branco/listpostsbyrating/repositories/PostRepositoryImpl.java

```diff
@@ -0,0 +1,114 @@
+package com.Joao_Branco.listpostsbyrating.repositories;
+
+import com.Joao_Branco.listpostsbyrating.Exceptions.NewPostException;
+import com.Joao_Branco.listpostsbyrating.domain.Post;
+import com.Joao_Branco.listpostsbyrating.domain.PostRowMapper;
+import com.Joao_Branco.listpostsbyrating.domain.PostSetExtractor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.ResultSetExtractor;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.util.Collection;
+
+@Repository
+public class PostRepositoryImpl implements PostRepository{
+
+    private static final String SQL_CREATE = "INSERT INTO posts(post_id, post_body, upvotes, downvotes, total_votes) VALUES(NEXTVAL('posts_seq'), ?, ?, ?, ?)";
+    private static final String SQL_FIND_BY_ID = "SELECT post_id, upvotes, downvotes, post_body, total_votes FROM posts WHERE post_id = ?";
+    private static final String SQL_GET_ALL_POSTS_ORDERED = "SELECT * FROM posts ORDER BY COALESCE(upvotes / NULLIF(total_votes,0), 0) DESC, COALESCE(downvotes / NULLIF(total_votes,0), 0) ASC, total_votes DESC";

```

Comment:
With this approach how will this order these:
Post 1: 601 upvotes and 399 downvotes
Post 2: 599 upvotes and 401 downvotes
Post 2: 600 upvotes and 400 downvotes
Post 4: 598 upvotes and 402 downvotes

Will it be consistent (always the same order) if you perform the list several times?

Reply:
And also, if it was requested, which orders could you implement with the information you have?

Reply:
I am not sure the ordering will be consistent if the list is performed several times. I just noticed now, after testing this case, that it seems to be returning the posts in an incorrect order. I hadn't considered this case when I was programming.